src/twisted/internet/abstract.py

```diff
@@ -269,27 +273,31 @@ def doWrite(self):
         if self.offset == len(self.dataBuffer) and not self._tempDataLen:
             self.dataBuffer = b""
             self.offset = 0
-            # stop writing.
-            self.stopWriting()
             # If I've got a producer who is supposed to supply me with data,
             if self.producer is not None and (
                 (not self.streamingProducer) or self.producerPaused
             ):
                 # tell them to supply some more.
```

Comment:
Maybe have a complete info

```suggestion
                # If I've got a producer who is supposed to supply me with data,
                # tell them to supply some more.
```

Reply:
I think I'm going to revert this, bit too worried about the consequences.